components/proxy/src/main/java/com/hotels/styx/infrastructure/FileBackedRegistry.java

```diff
@@ -35,95 +38,96 @@
  */
 public class FileBackedRegistry<T extends Identifiable> extends AbstractRegistry<T> {
     private static final Logger LOG = getLogger(FileBackedRegistry.class);
-    private final Parser<T> parser;
-    private final FileMonitor fileMonitor;
-
-    public FileBackedRegistry(Resource configurationFile, Parser<T> parser) {
-        this(parser, configurationFile.absolutePath(), path -> {
-            try {
-                return toByteArray(configurationFile.inputStream());
-            } catch (IOException e) {
-                throw propagate(e);
-            }
-        });
+    private final Resource configurationFile;
+    private final Reader<T> reader;
+    private HashCode fileHash;
+
+    public FileBackedRegistry(Resource configurationFile, Reader<T> reader) {
+        super("FileBackedRegistry");
+        this.configurationFile = checkNotNull(configurationFile);
+        this.reader = checkNotNull(reader);
+        this.fileHash = md5(configurationFile);
     }
 
-    FileBackedRegistry(Parser<T> parser, String absolutePath, Function<String, byte[]> byteLoader) {
-        this.fileMonitor = new FileMonitor(byteLoader, absolutePath);
-        this.parser = requireNonNull(parser);
+    private HashCode md5(Resource resource) {
+        try {
+            return Files.hash(new File(resource.absolutePath()), Hashing.md5());
+        } catch (IOException e) {
+            throw propagate(e);
+        }
     }
 
     @Override
-    protected void startUp() {
-        LOG.info("starting {}", getClass().getSimpleName());
-
-        fileMonitor.load();
+    protected CompletableFuture<Void> startService() {
+        return CompletableFuture.runAsync(() -> {
+            LOG.info("starting {}", getClass().getSimpleName());
 
-        Iterable<T> resources = parser.read(fileMonitor.bytes());
-        snapshot.set(resources);
-        Changes<T> changes = new Changes.Builder<T>()
-                .added(resources)
-                .build();
-        notifyListeners(changes);
+            Iterable<T> resources = reader.read(readFile());
+            snapshot.set(resources);
+            Changes<T> changes = new Changes.Builder<T>()
+                    .added(resources)
+                    .build();
+            notifyListeners(changes);
+        });
     }
 
     @Override
     public synchronized void reload(ReloadListener listener) {
-        if (!fileMonitor.load()) {
-            LOG.info("Not reloading {} as content did not change", fileMonitor.path());
+        if (!hasFileContentChanged()) {
```

Comment:
This code no longer reloads the file! The file loading should not have been changed anyway, since it is out of the scope of changing how services work in Styx.

Reply:
I need to look into this. Looks like unfinished work.

Reply:
But the original FileBackedRegistry was in a need of refactoring. Therefore the changes.

Reply:
It still reloads the file. This file had to be refactored. However, I noticed that parts of the code had evolved since rebasing this work, so I have taken liberties to refactor bit more.

I have removed a reload listener in favour of completable future.